File: extractor/repositories/example/python/example_code.py
from abc import ABC, abstractmethod
from x import y

class Vehicle(ABC):
    def __init__(self, name):
        self.name = name
        self.value

    @abstractmethod
    def run(self):
        pass

class Car(Vehicle):
    def run(self):
    	class Runner:
        	pass
        return "Car is running"

@decorate
class Motorcycle(Vehicle):
    def run(self):
        return "Motorcycle is running"

@decorator1
@decorator2(arg1, arg2)
@decorator3.decorate()
@decorator3.decorate().decorate2(arg1)
def run(vehicle1, vehicle2):
    x = 5
    print('start')
    vehicle1.run()
    print('stop')

vehicle = Car()
run(vehicle)

File: extractor/repositories/example/java/example_code.java
abstract class Vehicle {
    abstract void run();
}

class Car extends Vehicle {
    @Override
    void run() {
        System.out.println("Car is running");
    }
}

class Motorcycle extends Vehicle {
    @Override
    void run() {
        System.out.println("Motorcycle is running");
    }
}

public class Main {
    static void run(Vehicle vehicle) {
        System.out.println("start");
        vehicle.run();
        System.out.println("stop");
    }

    public static void main(String[] args) {
        Vehicle vehicle = new Car();
        run(vehicle);
    }
}